client/app/telecom/telephony/alias/special/rsva/telecom-telephony-alias-special-rsva.controller.js

```diff
@@ -0,0 +1,242 @@
+    this.tariffBearingPrice = `${this.costs.rsva.tariffBearing.value} ${(this.costs.rsva.tariffBearing.currencyCode === 'EUR' ? '€' : this.costs.rsva.tariffBearing.currencyCode)}`;
+
+    this.isLoading = true;
+
+    // Initialize bulk datas
+    this.bulkDatas = {
+      infos: {
+        name: 'rsva',
+        actions: [{
+          name: 'rsvaUpdate',
+          route: '/telephony/{billingAccount}/rsva/{serviceName}',
+          method: 'PUT',
+          params: null,
+        }, {
+          name: 'scheduleRateCode',
+          route: '/telephony/{billingAccount}/rsva/{serviceName}/scheduleRateCode',
+          method: 'POST',
+          params: null,
+        }],
+      },
+    };
+
+    return this.$q.all([
+      this.OvhApiTelephony.Rsva().v6().getCurrentRateCode({
+        billingAccount: this.billingAccount,
+        serviceName: this.serviceName,
+      }).$promise.then((rateCodeInfos) => {
+        this.rateCodeInfos = rateCodeInfos;
+        return this.OvhApiTelephony.Rsva().v6().getAllowedRateCodes({
+          billingAccount: this.billingAccount,
+          serviceName: this.serviceName,
+        }).$promise.then((allowedRateCodes) => {
+          this.allowedRateCodes = allowedRateCodes;
+          this.rsva.rateCode = _.find(this.allowedRateCodes, { code: _.get(rateCodeInfos, 'rateCode') });
+          this.rsvaForm.rateCode = this.rsva.rateCode;
+        });
+      }),
+
+      this.OvhApiTelephony.Rsva().v6().getScheduledRateCode({
+        billingAccount: this.billingAccount,
+        serviceName: this.serviceName,
+      }).$promise.then((scheduledRateCode) => {
+        if (scheduledRateCode) {
+          this.scheduledRateCode = scheduledRateCode;
+          this.formattedEffectiveDatetime = this.$filter('date')(this.scheduledRateCode.effectiveDatetime, 'dd/MM/yyyy');
+          this.formattedCancelLimitDatetime = this.$filter('date')(this.scheduledRateCode.cancelLimitDatetime, 'dd/MM/yyyy');
+          this.tariffBearingPrice = this.scheduledRateCode.updateRateCodePriceWithoutTax.value
+            ? this.scheduledRateCode.updateRateCodePriceWithoutTax.text : this.tariffBearingPrice;
+        }
+      }).catch(() => false),
+
+      this.OvhApiTelephonyService.v6().directory({
+        billingAccount: this.billingAccount,
+        serviceName: this.serviceName,
+      }).$promise.then((result) => {
+        this.rsvaInfos = _.pick(result, ['siret', 'email']);
+        return this.OvhApiTelephony.Rsva().v6().get({
+          billingAccount: this.billingAccount,
+          serviceName: this.serviceName,
+        }).$promise.then(details => this.OvhApiOrder.v6().schema().$promise.then((schema) => {
+          if (_.has(schema, 'models[\'telephony.NumberSpecialTypologyEnum\'].enum')) {
+            this.typologies = _.map(
+              _.filter(schema.models['telephony.NumberSpecialTypologyEnum'].enum, elt => elt.match(new RegExp(`^${result.country}_`))),
```

Comment:
Not sure it's better but you could use a variable for the `RegExp`, it will avoid recreating it for each element